doc/source/whatsnew/v3.0.0.rst

```diff
@@ -201,6 +201,67 @@ In cases with mixed-resolution inputs, the highest resolution is used:
     In [2]: pd.to_datetime([pd.Timestamp("2024-03-22 11:43:01"), "2024-03-22 11:43:01.002"]).dtype
     Out[2]: dtype('<M8[ns]')
 
+.. _whatsnew_300.api_breaking.value_counts_sorting:
+
+Changed behavior in ``value_counts`` when ``sort=False``
```

Comment:
```suggestion
Changed behavior in ``DataFrame.value_counts`` when ``sort=False``
```

(I think it is good to emphasize that this is about the DataFrame variant (because I assume most people only use the Series one) by also putting this in the section title)

Reply:
I think this is misleading because this also impacts `DataFrameGroupBy`.

Reply:
I made it both `DataFrame.value_counts` and `DataFrameGroupBy.value_counts` to the title of the subsection.

Reply:
Looks good!